### PR TITLE
adds functionality for mute without slew

### PR DESCRIPTION
**Pull request by curtis-j-campbell**
PR for adding allowing the mute block within a sigma flow to be toggled via this library

### CODE REVIEW

**Comment by MCUdude** [comment]
Thanks for the PR. I don't have a computer with SigmaStudio nearby, so can you post a screenshot of the block so I know what it looks like?

File: src/SigmaDSP.cpp [hunk 1]
```diff
@@ -180,6 +180,24 @@ void SigmaDSP::volume(uint16_t startMemoryAddress, float dB)
...
 }
 
+/**
+ * @brief control mute/unmute block without slew
+ *
+ * @param startMemoryAddress DSP memory address
+ * @param flag true or false
+ */
+void SigmaDSP::mute(uint16_t startMemoryAddress, bool flag)
+{
```
Comment (MCUdude) [inline src/SigmaDSP.cpp:190]: Please use the same indentation style as elsewhere in the library. Two white spaces, not four.

**Comment by curtis-j-campbell** [comment]
Hi, this is the block being controlled. I shouldve made a PR a few months ago but only just realised I hadnt. Would be good to get a review around the logic within `src/SigmaDSP.cpp` when possible.
image

File: src/SigmaDSP.cpp [hunk 2]
```diff
@@ -180,6 +180,24 @@ void SigmaDSP::volume(uint16_t startMemoryAddress, float dB)
...
+    if (flag)
```
Comment (MCUdude) [inline src/SigmaDSP.cpp:191]: `8388608` looks like a "magic number". Please use 0x800000 instead, like I've done other places in the code.
Also, `flag` doesn't say much. Use `state` (as in mute state) instead.

And I personally think the code is a bit verbose:
This is what I would prefer:
```cpp
/**
  * @brief Control mute/unmute block without slew
  *
  * @param startMemoryAddress DSP memory address
  * @param state true or false
  */
void SigmaDSP::mute(uint16_t startMemoryAddress, bool flag)
{
  safeload_write(startMemoryAddress, state ? 0x800000 : 0);
}
```

File: src/SigmaDSP.h [hunk 3]
```diff
@@ -74,6 +74,7 @@ class SigmaDSP
 
     void volume_slew(uint16_t startMemoryAddress,   float dB, uint8_t slew = 12);
     void volume(uint16_t startMemoryAddress,        float dB);
+    void mute(uint16_t startMemoryAddress, bool flag);
```
Comment (MCUdude) [inline src/SigmaDSP.h:77]: The formatting is a bit off here. Please align with the other arguments. And please use `state` instead of `flag`.

**Review by MCUdude — CHANGES_REQUESTED** [review]
I have a few comments you should look at

File: src/SigmaDSP.cpp (hunk 2 above)
Comment (curtis-j-campbell) [reply src/SigmaDSP.cpp]: Cheers for the feedback on this, completely forgot about ternary expressions. I've resolved this comments now

**Comment by MCUdude** [comment]
Thanks!

File: src/SigmaDSP.cpp (hunk 2 above)
Comment (MCUdude) [reply src/SigmaDSP.cpp]: No worries! Even though long and nested ternary expressions are difficult to read, they're neat for small and obvious things like flags!

**Pull request merged**